### PR TITLE
FIX: [droid] rework storage provider

### PR DESCRIPTION
**Pull request by koying**
Fixes Shield. Should preferably go for beta1
/cc @Montellese @MartijnKaijser

### CODE REVIEW

File: xbmc/storage/android/AndroidStorageProvider.cpp [hunk 1]
```diff
@@ -29,11 +29,29 @@
...
 #include "utils/URIUtils.h"
 
+static const char * typeWL[] = { "vfat", "exfat", "sdcardfs", "fuse", "ntfs", "fat32", "ext3", "ext4", "esdfs" };
+static const char * mountWL[] = { "/mnt", "/Removable", "/storage" };
+static const char * mountBL[] = {
+  "/mnt/secure",
+  "/mnt/shell",
+  "/mnt/asec",
+  "/mnt/obb",
+  "/mnt/media_rw/extSdCard",
```
Comment (BMc08GT) [inline xbmc/storage/android/AndroidStorageProvider.cpp]: may also want to add /mnt/media_rw/usbdisk* for OTG

Comment (koying) [inline xbmc/storage/android/AndroidStorageProvider.cpp]: To the blacklist?

Comment (BMc08GT) [inline xbmc/storage/android/AndroidStorageProvider.cpp]: Yes otg is mounted through fuse as well (mountpoint in media_rw and then fuse service creates the /storage/usbdisk)

Comment (koying) [inline xbmc/storage/android/AndroidStorageProvider.cpp]: ok

**Comment by koying** [comment]
jenkins build and merge

**Pull request merged**